metrics/prometheus/prometheus_test.go

```diff
@@ -8,80 +8,121 @@ import (
+
+	emptyResettingTimer.Update(time.Second) // no effect because of snapshot below
+	register(t, "test/empty_resetting_timer_snapshot", emptyResettingTimer.Snapshot())
+
+	// Nil metrics
+	metrics.Enabled = false
+	nilCounter := metrics.NewCounter()
+	register(t, "nil/counter", nilCounter)
+	nilCounterFloat64 := metrics.NewCounterFloat64()
+	register(t, "nil/counter_float64", nilCounterFloat64)
+	nilEWMA := &metrics.NilEWMA{}
+	register(t, "nil/ewma", nilEWMA)
+	nilGauge := metrics.NewGauge()
+	register(t, "nil/gauge", nilGauge)
+	nilGaugeFloat64 := metrics.NewGaugeFloat64()
+	register(t, "nil/gauge_float64", nilGaugeFloat64)
+	nilGaugeInfo := metrics.NewGaugeInfo()
+	register(t, "nil/gauge_info", nilGaugeInfo)
+	nilHealthcheck := metrics.NewHealthcheck(nil)
+	register(t, "nil/healthcheck", nilHealthcheck)
+	nilHistogram := metrics.NewHistogram(nil)
+	register(t, "nil/histogram", nilHistogram)
+	nilMeter := metrics.NewMeter()
+	register(t, "nil/meter", nilMeter)
+	nilResettingTimer := metrics.NewResettingTimer()
+	register(t, "nil/resetting_timer", nilResettingTimer)
+	nilSample := metrics.NewUniformSample(1028)
+	register(t, "nil/sample", nilSample)
+	nilTimer := metrics.NewTimer()
+	register(t, "nil/timer", nilTimer)
+
+	metrics.Enabled = true
+
+	gatherer := NewGatherer(registry)
+
+	families, err := gatherer.Gather()
+	require.NoError(t, err)
+	familyStrings := make([]string, len(families))
+	for i := range families {
+		familyStrings[i] = families[i].String()
+	}
+	want := []string{
+		`name:"test_counter" type:COUNTER metric:<counter:<value:12345 > > `,
+		`name:"test_counter_float64" type:COUNTER metric:<counter:<value:1.1 > > `,
+		`name:"test_gauge" type:GAUGE metric:<gauge:<value:23456 > > `,
+		`name:"test_gauge_float64" type:GAUGE metric:<gauge:<value:34567.89 > > `,
+		`name:"test_histogram" type:SUMMARY metric:<summary:<sample_count:0 sample_sum:0 quantile:<quantile:0.5 value:0 > quantile:<quantile:0.75 value:0 > quantile:<quantile:0.95 value:0 > quantile:<quantile:0.99 value:0 > quantile:<quantile:0.999 value:0 > quantile:<quantile:0.9999 value:0 > > > `,
+		`name:"test_meter" type:GAUGE metric:<gauge:<value:9.999999e+06 > > `,
+		`name:"test_resetting_timer" type:SUMMARY metric:<summary:<sample_count:1 quantile:<quantile:50 value:1e+09 > quantile:<quantile:95 value:1e+09 > quantile:<quantile:99 value:1e+09 > > > `,
+		`name:"test_timer" type:SUMMARY metric:<summary:<sample_count:6 sample_sum:2.3e+08 quantile:<quantile:0.5 value:2.25e+07 > quantile:<quantile:0.75 value:4.8e+07 > quantile:<quantile:0.95 value:1.2e+08 > quantile:<quantile:0.99 value:1.2e+08 > quantile:<quantile:0.999 value:1.2e+08 > quantile:<quantile:0.9999 value:1.2e+08 > > > `,
+	}
+	assert.Equal(t, want, familyStrings)
+
+	register(t, "unsupported", metrics.NewGaugeInfo())
+	families, err = gatherer.Gather()
+	assert.EqualError(t, err, "metric \"unsupported\": type is not supported: *metrics.StandardGaugeInfo")
```

Comment:
Use a common error and `errors.Is()` as this is a change-detector test for the error message.